### PR TITLE
In this branch I : Add reducers and actions

### PR DESCRIPTION
**Pull request by MohammedElseddik**
- Add react-redux package.
- Structure the project's files and follow the ducks-pattern.
- Define books Actions creator and type, and its reducer function.
- Define categories Actions creator and type, and its reducer function.
- Combine books and categories reducers into one root reducer.
- Create the store for the states of the bookStore application.

### CODE REVIEW

File: src/components/Books/Books.js [hunk 1]
```diff
@@ -25,7 +28,6 @@ const Books = () => {
 
   return (
     <div className="Books">
```
Comment (brytebee) [inline src/components/Books/Books.js:30]: - [x] **Optional:** Good job rendering the books. It is cleaner to use [`Object.map((blockVar) => {<Book/>})`](https://reactjs.org/docs/lists-and-keys.html) function to do this kind of rendering. You could check out t[he resource](https://reactjs.org/docs/lists-and-keys.html) if you consider implementing it.

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const CATEGORIES_STATUS = 'CATEGORY_STATUS';
```
Comment (brytebee) [inline src/redux/categories/categories.js]: - [ ] **Required:** Kindly use `ducks pattern` in defining this action type just as you rightly implemented in the books reducer file for `ADD_BOOK` and `REMOVE_BOOK`.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @MohammedElseddik👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- The structure of the files follows the `ducks pattern`.✔️
- Nicely implemented the Redux store.✔️
- Reducers work correctly.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @MohammedElseddik, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**